src/metabase/models/dashboard_card.clj

```diff
@@ -217,6 +221,117 @@
       (db/delete! DashboardCard :id (:id dashboard-card)))
     (events/publish-event! :dashboard-remove-cards {:id id :actor_id user-id :dashcards [dashboard-card]})))
 
+;;; ----------------------------------------------- Link cards ----------------------------------------------------
+
+(def ^:private all-card-info-columns
+  {:model         :text
+   :id            :integer
+   :name          :text
+   :description   :text
+
+   ;; for cards and datasets
+   :collection_id :integer
+   :display       :text
+
+   ;; for tables
+   :db_id        :integer})
+
+(def ^:private  link-card-columns-for-model*
+  {:database  [:id :name :description]
+   :table     [:id [:display_name :name] :description :db_id]
+   :dashboard [:id :name :description :collection_id]
+   :card      [:id :name :description :collection_id :display]
+   :dataset   [:id :name :description :collection_id :display]})
+
+(defn- ->column-alias
+  "Returns the column name. If the column is aliased, i.e. [`:original_name` `:aliased_name`], return the aliased
+  column name"
+  [column-or-aliased]
+  (if (sequential? column-or-aliased)
+    (second column-or-aliased)
+    column-or-aliased))
+
+(defn- link-card-columns-for-model
+  [model]
+  (let [model-cols-set                   (link-card-columns-for-model* model)
```

Comment:
nit: it's implied that model columns are unique, you don't need to call it a set explicitly. It's not even a clojure set anyway
```suggestion
  (let [model-cols                   (link-card-columns-for-model* model)
```

Reply:
my bad, I originally use link-card-columns-for-model* as a set, it should be removed.